gpkitmodels/Commercial Sizing/CFP_Fuselage_Performance.py

```diff
@@ -0,0 +1,978 @@
+                Vdb    == Adb*lshell,
+                #TODO Revert to posynomial after debugging
+                SignomialEquality(Vcabin, Afuse*(lshell + 0.67*lnose + 0.67*Rfuse)), #[SP] #[SPEquality]
+
+                # Weight relations
+                Wapu     == Wpay*fapu,
+                Wdb      == rhoskin*g*Vdb,
+                Winsul   >= Wppinsul*((1.1*pi+2*thetadb)*Rfuse*lshell + 0.55*(Snose+Sbulk)),
+                Wlugg    >= flugg2*npass*2*Wchecked + flugg1*npass*Wchecked + Wcarryon,
+                Wwindow  >= Wpwindow*lshell,
+                Wpadd    == Wpay*fpadd,
+                Wseat    == Wpseat*nseat,
+
+                Wskin    >= rhoskin*g*(Vcyl + Vnose + Vbulk),
+                Wshell   >= Wskin*(1 + fstring + ffadd + fframe) + Wdb, #+ Whbend, #+ Wvbend,
+                Wfuse       >= Wshell + Wfloor + Wtail + Winsul + Wapu + Wfix + Wwindow + Wpadd + Wseat + Whbend
+                ])
+
+        Model.__init__(self, None, constraints + self.wingbox)
+
+    def dynamic(self, state):
+        """
+        returns a fuselage performance model
+        """
+        return FuselagePerformance(self, state)
+
+class FuselagePerformance(Model):
+    """
+    Fuselage performance model
+    """
+    def __init__(self, fuse, state, **kwargs):
+        #new variables
+        Cdfuse = Variable('C_{D_{fuse}}', '-', 'Fuselage Drag Coefficient')
+        Dfuse    = Variable('D_{fuse}', 'N', 'Total drag in cruise')
+        Dfrict   = Variable('D_{friction}', 'N', 'Friction drag')
+        Dupswp   = Variable('D_{upsweep}', 'N', 'Drag due to fuse upsweep')
+        f        = Variable('f', '-', 'Fineness ratio')
+        FF       = Variable('FF', '-','Fuselage form factor')
+        phi      = Variable('\\phi', '-', 'Upsweep angle')
+
+        constraints = []
+        constraints.extend([
+            #Dfuse == Cdfuse * (.5 * fuse['A_{fuse}'] * state.atm['\\rho'] * state['V']**2),
+            f == fuse['l_{fuse}']/((4/np.pi*fuse['A_{fuse}'])**0.5), # fineness ratio
+            FF >= 1 + 60/f**3 + f/400, # form factor
+            Dfrict >= FF * np.pi*fuse['R_{fuse}']*state.atm['\\mu']*state['V']* 0.074*(state.atm['\\rho']*state['V']
+                                            *fuse['l_{fuse}']/state.atm['\\mu'])**0.8,
+            1.13226*phi**1.03759 == fuse['R_{fuse}']/fuse['l_{cone}'], # monomial fit of tan(phi)
+            Dupswp >= 3.83*phi**2.5*fuse['A_{fuse}']*0.5*state.atm['\\rho']*state['V']**2,
+            Dfuse >= Dfrict + Dupswp,
+            Dfuse == 0.5*state.atm['\\rho']*state['V']**2*Cdfuse*fuse['A_{fuse}']
+            ])
+
+        Model.__init__(self, None, constraints)
+    
+
+class Mission(Model):
+    """
+    mission class, links together all subclasses
+    """
+    def __init__(self, subs = None, **kwargs):
+        #define the number of each flight segment
+        Nclimb = 2
+        Ncruise = 2
+
+        #build required submodels
+        ac = Aircraft()
```

Comment:
it could be, would make the mission class more easily ported between different aircraft